src/chartdata/ChartData.js

```diff
@@ -0,0 +1,879 @@
+     * @return {ChartData} A new `ChartData` containing the mapped rows.
+     *
+     * @example
+     * const chartData = new ChartData(rows, columns);
+     * return chartData.mapRows(ii => ii * 2);
+     *
+     * @name mapRows
+     * @kind function
+     * @inner
+     * @memberof ChartData
+     */
+
+    mapRows(mapper: RowMapper): ChartData {
+        return new ChartData(this.rows.map(mapper), this.columns);
+    }
+
+    /**
+     * Returns all the data in a single column.
+     *
+     * @param {string} columns The name of the column.
+     * @return {?List<ChartScalar>} A list of the data, or null if columns have been set but the
+     * column could not be found.
+     *
+     * @example
+     * const chartData = new ChartData(rows, columns);
+     * return data.getColumnData('fruit');
+     * // returns List("apple", "apple", "orange", "peach", "pear")
+     *
+     * @name getColumnData
+     * @kind function
+     * @inner
+     * @memberof ChartData
+     */
+
+    getColumnData(column: string): List<ChartScalar> {
+        if(this._columnError(column)) {
+            return null;
+        }
+        return this._memoize(`getColumnData.${column}`, (): ?List<ChartScalar> => {
+            return this.rows.map(row => row.get(column));
+        });
+    }
+
+    /**
+     * For a given column, this returns a `List` of unique values in that column, in the order that
+     * each unique value first appears in `rows`.
+     *
+     * You can also return the number of unique values by calling `getUniqueValues().size`.
+     *
+     * @param {string} column The name of the column.
+     *
+     * @return {List<ChartScalar>|null}
+     * A `List` of unique values in the order they appear in `rows`, or null if columns have been
+     * set but the column could not be found.
+     *
+     * @name getUniqueValues
+     * @kind function
+     * @inner
+     * @memberof ChartData
+     */
+
+    getUniqueValues(column: string): ?List<ChartScalar> {
+        if(this._columnError(column)) {
+            return null;
+        }
+        return this._memoize(`getUniqueValues.${column}`, (): ?List<ChartScalar> => {
+            return this.rows
+                .map(ii => ii.get(column))
+                .toOrderedSet()
+                .toList();
+        });
+    }
+
+    /**
+     * This breaks `rows` data into frames, which are rows grouped by unique values of a
+     * specifed `frameColumn`.
+     *
+     * This method assumes that rows in the `frameColumn` are already sorted in the correct order.
+     *
+     * @param {string} column The name of the column.
+     *
+     * @return {List<List<ChartRow>>|null}
+     * A `List` of frames, where each frame is a `List`s of chart rows, or null if columns have been
+     * set but the column could not be found.
+     *
+     * @name makeFrames
+     * @kind function
+     * @inner
+     * @memberof ChartData
+     */
+
+    makeFrames(column: string): List<List<ChartRow>> {
```

Comment:
It would be cool if this could accept a function as well as a column string. That way if you have for example a bunch of data points with dates and you want to group them into months you can do something like.

```
makeFrames((row) => moment(row.get('date')).format('MM'));
```
To group them into months.